docs/adding-new-languages.md

```diff
@@ -0,0 +1,20 @@
+# Adding new languages to Semantic
+
+This document exists to outline the process associated with adding a new language to Semantic. Though the Semantic authors have architected the library such that adding new languages and syntax [requires no changes to existing code](https://en.wikipedia.org/wiki/Expression_problem), adding support for a new language is a nontrivial amount of work. Those willing to take the plunge will probably need a degree of Haskell experience.
```

Comment:
I'd add another (short) paragraph here calling out that this is temporary.  I like the detail you have below about TH/Core as a replacement (and that can stay down there), but it's worth mentioning that here at the beginning too.  Just a sentence or two, plus maybe a link to the more detailed discussion down in the FAQ.